pkg/check/mysql_server.go

```diff
@@ -33,8 +33,22 @@ func NewMySQLVersionChecker(db *sql.DB, dbinfo *dbutil.DBConfig) Checker {
 	return &MySQLVersionChecker{db: db, dbinfo: dbinfo}
 }
 
-// MinVersion is mysql minimal version required
-var MinVersion = [3]uint{5, 5, 0}
+// SupportedVersion defines the MySQL/MariaDB version that DM/syncer supports
+// * 5.6.0 <= MySQL Version < 5.8.0
+// * 10.1.2 <= Mariadb Version
+var SupportedVersion = map[string]struct {
+	Min MySQLVersion
+	Max MySQLVersion
+}{
+	"mysql": {
+		MySQLVersion{5, 6, 0},
+		MySQLVersion{5, 8, 0},
+	},
+	"mariadb": {
+		MySQLVersion{10, 1, 2},
+		UnlimitVersion,
```

Comment:
sugest add a MaxVersion = {math.MaxUint8,math.MaxUint8,math.MaxUint8}
and MinVersion = {0,0,0} if need
don't specially handle math.MaxUint8 in all compare func

Reply:
it also uses `UnlimitVersion` to indicate lowest boundary

Reply:
ok, I just want to save a variable